### ISSUE TITLE
Issues with escaped utf8 file names ?

### ISSUE THREAD

**Issue by Russsgithub** [aws/aws-cli#6372]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [ x] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [x ] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution
  
**Describe the bug**
I've been having problems with files with names containing non ascii characters, such as [this](Bára_Gísladóttir_and_Skúli_Sverrisson_-_Improvisation_3.mp3)

```
Bára_Gísladóttir_and_Skúli_Sverrisson_-_Improvisation_3.mp3
```

using  [liquidsoap](https://github.com/savonet/liquidsoap), which downloads files from s3 using awscli (aws).

Liquidsoap executes this command line:

```
aws s3 cp "s3://<bucket>/<filename>" "<local filename>"
```

to download the file. and say :

"We are escaping the utf8 file name following the patterns described in this PR: [#1839](https://github.com/savonet/liquidsoap/pull/1839)".

There is also this, that may be relavant. [https://stackoverflow.com/questions/27366479/python-3-os-walk-file-paths-unicodeencodeerror-utf-8-codec-cant-encode-s](https://stackoverflow.com/questions/27366479/python-3-os-walk-file-paths-unicodeencodeerror-utf-8-codec-cant-encode-s)

If I use awscliv1 all goes well, but using v2 I get this as an error:

```
fatal error: 'utf-8' codec can't encode characters in position 27-28: surrogates not allowed
```

[here](https://github.com/savonet/liquidsoap/discussions/1863) is a link to the discussion over at liquidsoaps git page.

**Comment by Russsgithub** [comment]
Hi @tim-finnigan . I'm getting this happen on an aws instance running Ubuntu Server 20.04.3, and a laptop running arch linux. As I mentioned the awscli is being called by [liquidsoap](https://www.liquidsoap.info/), their git page is [here](https://github.com/savonet/liquidsoap). The error is occurring with v2.0 of liquidsoap which is unreleased (code is downloadable from github and installed with OPAM), I'm not sure it occurs in the released version 1.4.4.

The issues you mention ...

The first seems to effect the cli from command line.. In my case calls from command line work fine.

The second seems closer, though none of the suggestions work in my case.

Setting locale to LC_ALL=en_US.UTF-8 did not solve the issue.

Thanks Tim

**Comment by tim-finnigan** [comment]
Hi @Russsgithub, sorry to hear you are still seeing this issue. But if your calls from the command line are working then I wonder how liquidsoap is executing the command differently. Can you provide debug logs to give move insight into what's going on?
